# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: xpcom/build/Omnijar.cpp
# Commit: 17a501a19918
# Full Hash: 17a501a1991842d8488ff3e7d4611bb688abfc87
# Author: Randell Jesup <rjesup@wgate.com>
# Date: 2022-03-24 21:54:20
# Regressor Bug: 1744043
# File Overlap Count: 1
# Description:
#   Bug 1744043: Clean up nsJAR r=nika,valentin,extension-reviewers,robwu
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D132794
# ==============================================================================

diff -r daed1e9ef335 -r 17a501a19918 xpcom/build/Omnijar.cpp
--- a/xpcom/build/Omnijar.cpp	Thu Mar 24 13:45:28 2022 +0000
+++ b/xpcom/build/Omnijar.cpp	Thu Mar 24 13:54:56 2022 +0000
@@ -26,11 +26,9 @@
 
 void Omnijar::CleanUpOne(Type aType) {
   if (sReader[aType]) {
-    sReader[aType]->CloseArchive();
     sReader[aType] = nullptr;
   }
   if (sOuterReader[aType]) {
-    sOuterReader[aType]->CloseArchive();
     sOuterReader[aType] = nullptr;
   }
   sPath[aType] = nullptr;
@@ -77,8 +75,8 @@
     return;
   }
 
-  RefPtr<nsZipArchive> zipReader = new nsZipArchive();
-  if (NS_FAILED(zipReader->OpenArchive(file))) {
+  RefPtr<nsZipArchive> zipReader = nsZipArchive::OpenArchive(file);
+  if (!zipReader) {
     return;
   }
 
@@ -87,8 +85,8 @@
   if (NS_SUCCEEDED(nsZipHandle::Init(zipReader, MOZ_STRINGIFY(OMNIJAR_NAME),
                                      getter_AddRefs(handle)))) {
     outerReader = zipReader;
-    zipReader = new nsZipArchive();
-    if (NS_FAILED(zipReader->OpenArchive(handle))) {
+    zipReader = nsZipArchive::OpenArchive(handle);
+    if (!zipReader) {
       return;
     }
   }
